### ISSUE TITLE
ESM `package.json` in home directory crashes LSP

### ISSUE THREAD

**Issue by Hacksore** [zbirenbaum/copilot.lua#306]
I was looking into a [Discord bug I found](https://x.com/Hacksore/status/1790395583611474423) and funnily enough it also breaks the copilot LSP 😂.

```
[START][2024-08-03 11:32:06] LSP logging initiated
[ERROR][2024-08-03 11:32:06] .../vim/lsp/rpc.lua:770	"rpc"	"node"	"stderr"	"file:///Users/hacksore/.local/share/nvim/lazy/copilot.lua/copilot/index.js:3
global.__rootDirectory = __dirname + '/dist/';
                         ^

ReferenceError: __dirname is not defined in ES module scope
This file is being treated as an ES module because it has a '.js' file extension and '/Users/hacksore/package.json' contains \"type\": \"module\". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///Users/hacksore/.local/share/nvim/lazy/copilot.lua/copilot/index.js:3:26
    at ModuleJob.run (node:internal/modules/esm/module_job:222:25)
    at async ModuleLoader.import (node:internal/modules/esm/loader:323:24)
    at async loadESM (node:internal/process/esm_loader:28:7)
    at async handleMainPromise (node:internal/modules/run_main:113:12)

Node.js v20.12.2
```

Not sure this is fixable but raising the issue for awareness just in case others run into this.

**tl;dr** don't have a `package.json` in your homedir with `"type": "module"` 💀.

**Comment by Hacksore** [comment]
Since this is really just for SEO nothing to do here unless someone has ideas. Closing for now!

**Comment by Hacksore** [comment]
Actually after some testing and the discussion in https://github.com/nodejs/node/issues/54632.

If we stick a non-module package.json here it should stop node from trying to traverse up finding an incorrect one.  `~/.local/share/nvim/lazy/copilot.lua/copilot/package.json` 

I'll get a PR up to solve this!
